### PR TITLE
[DbtProject] Rename `--use-experimental-dbt-project` flag to `--use-dbt-project` in dbt scaffold CLI

### PR DESCRIPTION
**Pull request by maximearmstrong**
## Summary & Motivation

This PR renames the `--use-experimental-dbt-project` flag to `--use-dbt-project` in the dbt scaffold CLI, now that [DbtProject is no more experimental](https://github.com/dagster-io/dagster/pull/22578). 

This is the first step in making DbtProject the default and only strategy when scaffolding a dbt project. The goal is to facilitate the transition from the previous default strategy to that of DbtProject in our tutorials, courses and docs.

Once this is merged, next steps will be to
- update tutorials, courses and docs to use DbtProject.
    - Since our examples heavily rely on the scaffold command, we'll be able to use `dagster-dbt project scaffold --use-dbt-project` with the updated flag
- make the DbtProject strategy the default and only one when using `dagster-dbt project scaffold`
- update tutorials, courses and docs to remove the flag

Tickets are listed [here](https://linear.app/dagster-labs/issue/DS-337/make-dbtproject-the-default-scaffold-strategy).

## How I Tested These Changes

BK with updated tests

### CODE REVIEW

**Comment by maximearmstrong** [comment]
* **#22794** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/22794?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @maximearmstrong and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: python_modules/libraries/dagster-dbt/dagster_dbt/cli/app.py [hunk 1]
```diff
@@ -246,11 +246,11 @@ def project_scaffold_command(
             hidden=True,
         ),
     ] = False,
-    use_experimental_dbt_project: Annotated[
+    use_dbt_project: Annotated[
         bool,
         typer.Option(
             ...,
-            "--use-experimental-dbt-project",
```
Comment (rexledesma) [inline python_modules/libraries/dagster-dbt/dagster_dbt/cli/app.py:253]: Can also just keep `--use-experimental-dbt-project` as a flag so things that use that flag won't break (e.g. extraneous docs, internal).

**Review by rexledesma — APPROVED** [review]
this also needs an internal PR

**Comment by maximearmstrong** [comment]
> this also needs an internal PR

@rexledesma It's ready for review [here](https://github.com/dagster-io/internal/pull/10425)

**Comment by rexledesma** [comment]
Sweet thanks

**Pull request merged**